experimental/Schemes/Resolution_structure.jl

```diff
@@ -0,0 +1,738 @@
+# test for snc                                                         #
+########################################################################
+
+function divisor_intersections_with_X(current_div, I_X)
+  scheme(I_X) == scheme(current_div[1]) || error("underlying schemes do not match")
+  n_max = dim(I_X)
+
+  inter_div_dict = Dict{Vector{Int},Tuple{AbsIdealSheaf,Int}}()
+  old_keys = Vector{Int}[]
+  empty_keys = Vector{Int}[]
+  essential_inter = AbsIdealSheaf[]
+
+# initialization: each divisor + I_X
+  for k in 1:length(current_div)
+    Idiv = ideal_sheaf(current_div[k]) + I_X
+    if !is_one(Idiv)
+      inter_div_dict[[k]] = (Idiv,0)
+      push!(old_keys, [k])
+    end
+  end
+  new_keys = copy(empty_keys)
+
+# add intersections
+  while !is_empty(old_keys)
+    old_keyvec = popfirst!(old_keys)         # this is the intersection to which we add a new divisor
+    for i in (old_keyvec[end]+1):length(current_div)
+      copykey = copy(old_keyvec)
+      if haskey(inter_div_dict, [i])
+        push!(copykey,i)                     # here we add it
+        Idiv = inter_div_dict[[i]][1] + inter_div_dict[old_keyvec][1]
+        if !is_one(Idiv)
+                                             # it intersects
+          subsetlist = subsets(old_keyvec,length(old_keyvec)-1)
+          subsetlist = [push!(a,i) for a in subsetlist]
+          if (sum([inter_div_dict[a][2] for a in subsetlist])> 0)
+                                             # offending intersection, known before
+            inter_div_dict[copykey] = (Idiv,2)
+            push!(new_keys,copykey)
+          elseif dim(Idiv) == n_max - length(copykey)
+                                             # non-offending intersection
+            inter_div_dict[copykey] = (Idiv,0)
+            push!(new_keys,copykey)
+          else
+                                             # offending intersection, new
+            inter_div_dict[copykey] = (Idiv,1)
+            push!(new_keys,copykey)
+            push!(essential_inter, Idiv)
+          end
+        end
+      end
+    end
+
+    # go to next higher number of intersecting divisors
+    if is_empty(old_keys)
+      old_keys = copy(new_keys)
+      new_keys = copy(empty_keys)
+    end
+  end
+
+  return inter_div_dict, essential_inter
+end
+
+function non_snc_locus(divs::Vector{<:EffectiveCartierDivisor})
+  is_empty(divs) && error("list of divisors must not be empty")
+  X = scheme(first(divs))
+  @assert all(d->scheme(d) === X, divs)
+  @assert is_smooth(X)
+  r = length(divs)
+  triv_cov = trivializing_covering.(divs)
+
+  com_ref, incs = common_refinement(triv_cov, default_covering(X))
+
+  ideal_dict = IdDict{AbsAffineScheme, Ideal}() # ideal sheaf of the non_snc_locus
+  for U in patches(com_ref)
+    loc_eqns = elem_type(OO(U))[]
+    for k in 1:length(incs)
+      I = ideal_sheaf(divs[k])
+      inc = incs[k][U]
+      V = codomain(inc)
+      h = first(gens(I(V)))
```

Comment:
This should avoid an allocation?
```suggestion
      h = gen(I(V), 1)
```

Reply:
done